HRTool/src/main/java/app/controllers/ReportRestController.java

```diff
@@ -129,22 +133,24 @@ else if(tc.getTipTagCloud().equals(TagCloudEnum.IDE)) {
     	if(file != null) {
     		file.delete();
     	}
-    	ResponseEntity<byte[]> response = new ResponseEntity<byte[]>(data, headers, HttpStatus.OK);
+    	ResponseEntity response = new ResponseEntity(data, headers, HttpStatus.OK);
         return response;
 	}
 	
+	@SuppressWarnings({ "rawtypes", "unchecked" })
 	@RequestMapping("/seniority")
 	/**
 	 * Sends seniority pie chart as an array of bytes.
 	 * @return Seniority pie chart
 	 */
-	public ResponseEntity<byte[]> generateChart() {
+	public ResponseEntity generateChart() {
+		File file = null;
 		try {
-			ChartGenerator.generatePieChart(employeeRepository.findAll());
+			file = ChartGenerator.generatePieChart(employeeRepository.findAll());
 		} catch (FileNotFoundException | com.itextpdf.text.DocumentException | ParseException e) {
 			e.printStackTrace();

```

Comment:
One does not simply printStackTrace() :)
Replace with logger. Return ResponseEntity to notifiy the client.